src/binder/inc/bindertracing.h

```diff
@@ -49,6 +50,61 @@ namespace BinderTracing
         ReleaseHolder<PEAssembly> m_resultAssembly;
         bool m_cached;
     };
+
+
+    class ResolutionAttemptedOperation
```

Comment:
Can you add a comment about the general usage of this class? e.g. when it would fire events, how/order in which the consumer is expected to call `Trace` and `SetResult`

Reply:
I'm actually a bit unclear myself now. From the usage, it seems like `Trace` gets called first and is expected to fire an event indicating the start of a stage, then `SetResult` is called, then the destructor will fire an event indicating the end of a stage (and its result). However, the same class is used for tracing multiple stages and I don't see how the end result of each stage is traced before the start of the next one?

Reply:
Rewrote the class and wrote a comment explaining how it works.